stub.sh

```diff
@@ -26,6 +26,29 @@
 # IN THE SOFTWARE.
 #
 
+# Global variables
+MAIN_SPID=$BASHPID
+STUB_FIFO=/tmp/stub_fifo
+
+# fifo initialization
+[ ! -p $STUB_FIFO ] && mkfifo $STUB_FIFO
+
+# trap setup
+trap __read_fifo USR1
+trap __fifo_clean_up EXIT
+
+
+# Private: Reads the fifo when the USR1 signal is received
+__read_fifo() {
+  read EXEC < $STUB_FIFO
+  [ -n "$EXEC" ] && eval $EXEC
+  return 0
+}
+
+# Private: Clean up fifo resources created at startup
+__fifo_clean_up() {
```

Comment:
sadly, this is not executed when sourcing the file so the named pipe is not always deleted, depending on the test suite

At the end of the day is a simple fd created at tmp that will eventually delete itself so I assume it's no big deal.

Let me know if I am missing something here!